pkg/skaffold/deploy/timings.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+)
+
+// WithTimings creates a deployer that logs the duration of each phase.
+func WithTimings(d Deployer) Deployer {
+	return withTimings{
+		Deployer: d,
+	}
+}
+
+type withTimings struct {
+	Deployer
+}
+
+func (w withTimings) Deploy(ctx context.Context, out io.Writer, builds []build.Build) error {
+	start := time.Now()
+	fmt.Fprintln(out, "Starting deploy...")
+
+	err := w.Deployer.Deploy(ctx, out, builds)
+	if err != nil {
+		return err
```

Comment:
I don't really like wrapping the error here since all we do is delegate to a function call without changing the inputs or outputs. It just adds noise to the errors I think.